Lib/pathlib.py

```diff
@@ -479,6 +260,113 @@ def __reduce__(self):
         # when pickling related paths.
         return (self.__class__, tuple(self._parts))
 
+    @classmethod
+    def _split_extended_path(cls, s):
+        prefix = ''
+        if s.startswith(_WIN_EXT_NAMESPACE_PREFIX):
+            prefix = s[:4]
+            s = s[4:]
+            if s.startswith('UNC\\'):
+                prefix += s[:3]
+                s = '\\' + s[3:]
+        return prefix, s
+
+    @classmethod
+    def _splitroot(cls, part):
+        # FIXME: This method should use os.path.splitdrive()
+        sep = cls._flavour.sep
+        if cls._flavour is posixpath:
+            if part and part[0] == sep:
+                stripped_part = part.lstrip(sep)
+                # According to POSIX path resolution:
+                # http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap04.html#tag_04_11
+                # "A pathname that begins with two successive slashes may be
+                # interpreted in an implementation-defined manner, although more
+                # than two leading slashes shall be treated as a single slash".
+                if len(part) - len(stripped_part) == 2:
+                    return '', sep * 2, stripped_part
+                else:
+                    return '', sep, stripped_part
+            else:
+                return '', '', part
+
+        first = part[0:1]
+        second = part[1:2]
+        if (second == sep and first == sep):
+            # XXX extended paths should also disable the collapsing of "."
```

Comment:
There shouldn't be an `XXX` comments left.

Reply:
Is that true even when moving code around? Fixing other bugs increases the overall risk of the patch.

Reply:
The comment is mistaken. The Windows API skips normalizing extended paths only when they're opened. `GetFullPathNameW()`, on the other hand, normalizes all UNC paths, including "\\\\.\\" and "\\\\?\\" paths. This includes stripping trailing dots and spaces from the final path component, which has yet to be implemented by either `ntpath.normpath()` or `pathlib`.

Currently `pathlib` doesn't support literal paths that skip normalization, such as allowing "." components and repeated separators. It doesn't even support an initial "." component such as "./spam", which is sometimes necessary in both POSIX and Windows.